sqlmesh/core/engine_adapter/snowflake.py

```diff
@@ -83,16 +84,14 @@ def _fetch_native_df(
             columns = self.cursor._result_set.batches[0].column_names
             return pd.DataFrame([dict(zip(columns, row)) for row in rows])
 
-    def _get_data_objects(
-        self, schema_name: str, catalog_name: t.Optional[str] = None
-    ) -> t.List[DataObject]:
+    @set_catalog(schema_field=(1, "schema_name"), override=CatalogSupport.REQUIRES_SET_CATALOG)
+    def _get_data_objects(self, schema_name: t.Union[str, exp.Table]) -> t.List[DataObject]:
         """
         Returns all the data objects that exist in the given schema and optionally catalog.
         """
         from snowflake.connector.errors import ProgrammingError
 
-        target = nullsafe_join(".", catalog_name, schema_name)
-        sql = f"SHOW TERSE OBJECTS IN {target}"
+        sql = f'SHOW TERSE OBJECTS IN "{exp.to_table(schema_name, dialect=self.dialect).sql(dialect=self.dialect)}"'
```

Comment:
I think neither the old approach or this approach properly deals with identifiers in Snowflake. Any advice on the best way to do this for Snowflake?